dnase/callsnpsMerge.sh

```diff
@@ -187,14 +187,47 @@ for sampleid in `bcftools query -l ${sampleOutdir}/${name}.${mappedgenome}.filte
     fi
     #If there is no database ID (e.g. rsid), set up bed ID as chrom : pos _ alt
     cat $TMPDIR/variants.${sampleid}.txt | awk -F "\t" 'BEGIN {OFS="\t"} $4=="." {split($8, gt, "/"); gtout=gt[1]; if(length(gt)>2) {gtout=gtout "/" gt[2]}; chromnum=$1; gsub(/^chr/, "", chromnum); $4=chromnum ":" $2+1 "_" gtout } {print}' | sort-bed - | starch - > ${sampleOutdir}/${name}${vcfsamplename}.${mappedgenome}.genotypes.starch
-    
+
+    # Assign colors to the genotypes.bb lines.
+    rm -f "${TMPDIR}/${name}${vcfsamplename}.${mappedgenome}.newBedColumns.bed"   # Cleaning up from previous run, since using ">>" in line 221 below.
+    while read -r line_in; do
+        read -r chrom chromStart chromEnd readID qual ref alt betterGT phredGenoLikelihoods numTotalReads numRefReads numAltReads <<< "${line_in}"
+
+        IFS='/' read -ra alleles <<< "${betterGT}"
+        numAlleles=${#alleles[@]}
+
+        if [ ${numAlleles} -eq 1 ]; then
+            # haploid - always homozygous
+            if [ "${alleles[0]}" = "${ref}" ]; then
+                colors="105,105,105"  # hz_ref - grey
+            else
+                colors="253,199,0"    # hz_nonref - yellow
+            fi
+        else
+            # diploid
+            if [ "${alleles[0]}" = "${alleles[1]}" ]; then
+                # homozygous
+                if [ "${alleles[0]}" = "${ref}" ]; then
+                    colors="105,105,105"  # hz_ref - grey
+                else
+                    colors="253,199,0"    # hz_nonref - yellow
+                fi
+            else
+                # heterozygous
+                colors="19,165,220"   # het - blue
+            fi
+        fi
+
```

Comment:
Can you translate this logic into awk? Then it can go in the awk command on line 193 and avoid paste below so there is a single data conduit.